src/plugins/CopyFileContents/index.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2024 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import "./style.css";
+
+import ErrorBoundary from "@components/ErrorBoundary";
+import { CopyIcon, NoEntrySignIcon } from "@components/Icons";
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { Clipboard, Tooltip, useState } from "@webpack/common";
+
+const CheckMarkIcon = () => {
+    return <svg aria-hidden="true" width="24" height="24" fill="none" viewBox="0 0 24 24">
+        <path fill="currentColor" d="M21.7 5.3a1 1 0 0 1 0 1.4l-12 12a1 1 0 0 1-1.4 0l-6-6a1 1 0 1 1 1.4-1.4L9 16.58l11.3-11.3a1 1 0 0 1 1.4 0Z"></path>
+    </svg>;
+};
+
+export default definePlugin({
+    name: "CopyFileContents",
+    description: "Adds a button to copy text file attachment contents.",
+    authors: [Devs.Obsidian, Devs.Nuckyz],
+    patches: [
+        {
+            find: ".Messages.PREVIEW_BYTES_LEFT.format(",
+            replacement: {
+                match: /\.footerGap.+?url:\i,fileName:\i,fileSize:\i}\),(?<=fileContents:(\i),bytesLeft:(\i).+?)/g,
+                replace: "$&$self.AddCopyButton({ fileContents: $1, bytesLeft: $2 }),"
```

Comment:
I dont think youre supposed to put spaces in replaces
```suggestion
                replace: "$&self.AddCopyButton({$1,$2}),"
```
verify that works, im not sure, and im tired